### PR TITLE
NF: add routine to read zero-terminal strings

### PR DESCRIPTION
**Pull request by matthew-brett**
Routine to read zero-terminal strings of arbitrary length from binary file
stream.  Inspired by Thomas Emmerling's readCstring function in the 
BrainVoyager PR.

### CODE REVIEW

File: nibabel/tests/test_fileutils.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        # create a tempfile
+        path = 'test.bin'
+        fwrite = open(path, 'wb')
+        # write the binary block to it
+        fwrite.write(binary)
+        fwrite.close()
+        # open it again
+        fread = open(path, 'rb')
+        # test readout of one string
+        assert_equal([s for s in read_zt_byte_strings(fread)],
```
Comment (thomastweets) [inline nibabel/tests/test_fileutils.py]: This should be possible without a list comprehension as the return value is no generator anymore, right?

**Comment by thomastweets** [comment]
Nice!
This would exactly fit my use case of parsing zt strings from BV file headers.
I wonder whether one should include a 'rewind' kind of parameter as the changed position of the fileobj could be a not-so-obvious side effect. Then again, maybe we can assume well informed users.

File: nibabel/tests/test_fileutils.py (hunk 1 above)
Comment (matthew-brett) [inline nibabel/tests/test_fileutils.py]: Yes, thanks for finding that one.

**Comment by matthew-brett** [comment]
I left out `rewind` because I thought the most common case is that you expect the file pointer to be after the stuff you have just read.  Also, it's easy to wrap this function with a few lines to do that if you need to.

I added a note about the file position in the docstring.

**Comment by thomastweets** [comment]
Perfect!

**Comment by matthew-brett** [comment]
Thanks for the review - merging...

**Pull request merged**